Mathlib/Algebra/Ring/Defs.lean

```diff
@@ -140,7 +140,7 @@ addition, and `0` and `1` are additive and multiplicative identities. -/
 class Semiring (α : Type u) extends NonUnitalSemiring α, NonAssocSemiring α, MonoidWithZero α
 
 /-- A `Ring` is a `Semiring` with negation making it an additive group. -/
-class Ring (R : Type u) extends Semiring R, AddCommGroup R, AddGroupWithOne R
+class Ring (α : Type u) extends Semiring α, NonUnitalRing α, NonAssocRing α
```

Comment:
Note that not doing this was originally a performance choice. I recommend you benchmark this separately from adding the new typeclasses.